lib/utilities.js

```diff
@@ -24,22 +24,29 @@ const captureMaxStack = () => {
 
 const nodeModuleMatch = /internal[/\\]modules[/\\](cjs|esm)[/\\]/;
 
+// Try to detect the filepath of a caller of this function.
+// Signature: [depth [, stack]]
+//   depth <number> | <RegExp> initial stack slice or filter regular expression,
+//       0 by default.
+//   stack <string> stack string, optional
 const callerFilepath = (depth = 0, stack = null) => {
-  if (!stack) {
-    const origLimit = Error.stackTraceLimit;
-    Error.stackTraceLimit = 50;
-    stack = new Error().stack;
-    Error.stackTraceLimit = origLimit;
+  if (!stack) stack = captureMaxStack();
+  // remove first 'Error' line, captureMaxStack and this function
+  stack = stack.split('\n').slice(3);
```

Comment:
This becomes a breaking change, if you use `slice(3)` here, since before when you passed the stack into the function, it removed first 2 lines, not 3, thus this may break existing usage of this function.

Reply:
Yep, fixed.